src/components/hand-controls.js

```diff
@@ -74,34 +126,95 @@ module.exports.Component = registerComponent('hand-controls', {
-      case 'trigger':
-        if (isPressed === this.triggerPressed) { return; }
-        this.triggerPressed = isPressed;
-        this.playAnimation('pointing', !isPressed);
-        evt = isPressed ? 'pointup' : 'pointdown';
-        el.emit(evt);
-        break;
-      case 'grip':
-        if (isPressed === this.gripPressed) { return; }
-        this.gripPressed = isPressed;
-        this.playAnimation('close', !isPressed);
-        evt = isPressed ? 'gripclose' : 'gripopen';
-        el.emit(evt);
-        break;
+    var isTouched = evt === 'touchstart';
+    if (evt.indexOf('touch') === 0) {
+      if (isTouched === this.touchedButtons[button]) { return; }
+      this.touchedButtons[button] = isTouched;
+    } else {
+      if (isPressed === this.pressedButtons[button]) { return; }
+      this.pressedButtons[button] = isPressed;
+    }
+    if (!this.determineGesture()) { return; }
+    this.animateGesture();
+    this.emitGestureEvents();
+  },
+
+  determineGesture: function () {
+    var lastGesture = this.gesture;
+    var gesture;
+    if (this.pressedButtons['grip']) {
+      if (this.pressedButtons['surface'] || this.touchedButtons['surface'] ||
+          this.touchedButtons['menu'] || this.touchedButtons['AorX'] ||
+          this.pressedButtons['trackpad'] || this.touchedButtons['trackpad']) {
+        if (!this.pressedButtons['trigger'] && !this.touchedButtons['trigger']) {
+          // pointing pose
+          gesture = 'pointing';
+        } else {
+          // make a fist
+          gesture = 'fist';
+        }
+      } else {
+        if (!this.pressedButtons['trigger'] && !this.touchedButtons['trigger']) {
+          // pistol pose
+          gesture = 'pistol';
+        } else {
+          // thumbs up
+          gesture = 'thumb';
+        }
+      }
+    } else {
+      // grip not pressed
+      if (!this.pressedButtons['trigger'] && !this.touchedButtons['trigger']) {
+        // TODO: seems as though we should have some additional poses here
+        // leave gesture as null
+      } else {
+        // touch pose (?)
+        gesture = 'touch';
+      }
+    }
+    this.gesture = gesture;
+    return lastGesture !== this.gesture;
+  },
+
+  gestureAnimationMapping: {
+    'pointing': 'pointing',
+    'pistol': 'pistol',
+    'fist': 'press',
+    'touch': 'touch',
+    'thumb': 'thumb'
+  },
+
+  animateGesture: function () {
+    var animation = this.gestureAnimationMapping[this.gesture || ''];
+    this.playAnimation(animation || 'touch', !animation);
+  },
+
+  // map to old vive-specific event names for now
+  gestureEventMapping: {
+    'fist': 'grip',  // e.g. grip button down
+    'touch': 'point', // e.g. trigger button down
+    'thumb': 'thumb'  // e.g. thumbs up pose - grip button down, trackpad / surface buttons up
+  },
+
+  emitGestureEvents: function () {
+    var gesture = this.gesture || '';
+    var lastGesture = this.lastGesture;
+    var eventName;
+    if (lastGesture !== gesture) {
```

Comment:
actually since we're moving change detection into handleButton, this should be passed the (new) gesture at least, if not the last gesture, right?  that way you can simulate gesture events if necessary by calling emitGestureEvents directly with appropriate parameters